axelor-crm/src/main/java/com/axelor/apps/crm/web/ConvertLeadWizardController.java

```diff
@@ -177,14 +177,14 @@ public void setDefaults(ActionRequest request, ActionResponse response) throws A
 
     Lead lead = findLead(request);
 
-    response.setAttr("$primaryAddress", "value", lead.getPrimaryAddress());
-    response.setAttr("$primaryCity", "value", lead.getPrimaryCity());
-    response.setAttr("$primaryState", "value", lead.getPrimaryState());
-    response.setAttr("$primaryPostalCode", "value", lead.getPrimaryPostalCode());
-    response.setAttr("$primaryCountry", "value", lead.getPrimaryCountry());
+    response.setAttr("$partner.$primaryAddress", "value", lead.getPrimaryAddress());
+    response.setAttr("$partner.$primaryCity", "value", lead.getPrimaryCity().getName());
+    response.setAttr("$partner.$primaryState", "value", lead.getPrimaryState().getName());
```

Comment:
if using reference dummy field no need of setting name. directly set the object.